src/ex1/script.js

```diff
@@ -0,0 +1,158 @@
+        alert("todo cannot be empty")
+        return
+    }
+
+    checkIfExistDataFromLS()
+    pushEnteredDataToLS(enterValue)
+    showTodos()
+
+    addTodoButton.classList.remove("active")
+}
+
+function checkIfExistDataFromLS(){
+    let dataFromLS = localStorage.getItem("new-todo")
+
+    if(dataFromLS === null){
+        todoList = []
+    }
+    else{
+        todoList = JSON.parse(dataFromLS)
+    }
+}
+
+function pushEnteredDataToLS(enterValue){
+    todoList.push(enterValue)
+    localStorage.setItem("new-todo", JSON.stringify(todoList))
+    alert(`added new todo ${enterValue}`)
+}
+
+function showTodos() {
+    checkIfExistDataFromLS()
+    showMatchUiByTodosNumber()
+    createTodoListItems()
+}
+
+function showMatchUiByTodosNumber() {
+    sumTodos.textContent = todoList.length
+    
+    if(todoList.length > 0){
+        clearAllTodosButton.classList.add("active")
+        clearAllTodosButton.style.cursor = "pointer"
+        enterTodos.style.display = "none"
+    }
+    else{
+        clearAllTodosButton.classList.remove("active")
+        clearAllTodosButton.style.cursor = "not-allowed"
+        enterTodos.style.display = "block"
+    }
+}
+
+function createTodoListItems() {
+    let listItems = ""
+    todoList.forEach((todo, index) => { 
+        listItems += `<li>${todo}
+            <span class="delete" onclick="deleteTodo(${index})";>
+                <i class="fas fa-trash"></i>
+            </span>
+            <span class="edit" onclick="editTodo(${index})";>
+                <i class="fas fa-pen"></i>
+            </span>
+            </li>
+            `
+    })
+
+    todoListElement.innerHTML = listItems
+
+    todoInput.value = ""
+}
+
+function deleteTodo(index) {
+    
+        let dataFromLS = localStorage.getItem("new-todo")
+        todoList = JSON.parse(dataFromLS)
+        const removedTodo = todoList[index]
+        todoList.splice(index, 1) //remove one todo
+        alert(`removed new todo ${removedTodo}`)
+        localStorage.setItem("new-todo", JSON.stringify(todoList))
+        showTodos()       
+}
+
+function editTodo(index) {
+    let editedValue = prompt('Edit todo:', todoList[index])
+    let dataFromLS = localStorage.getItem("new-todo")
+    todoList = JSON.parse(dataFromLS)
+    todoList[index] = editedValue
+    alert(`edited todo ${editedValue}`)
+    localStorage.setItem("new-todo", JSON.stringify(todoList))
+    showTodos()
+}
+
+clearAllTodosButton.addEventListener("click", () => {
+    let dataFromLS = localStorage.getItem("new-todo")
+
+    if(dataFromLS === null){
+        todoList = []
+    }
+    else{
+        todoList = JSON.parse(dataFromLS)
+        todoList = [] //initialize array again
+    }
+
+    alert('removed all todos')
+    localStorage.setItem("new-todo", JSON.stringify(todoList))
+    showTodos()
+})
+
+orderSelect.addEventListener('change', (e) => {
+    let dataFromLS = localStorage.getItem("new-todo")
+    todoList = JSON.parse(dataFromLS)
```

Comment:
it's better to encapsulate all the local storage operations behind functions to avoid accessing it directly. Then you could also build wrappers that both fetch and save the todos, and handle that calls to the local storage functions when needed